### ISSUE TITLE
Why does the first one generate doc and the second one not

### ISSUE THREAD

**Issue by xiaxichen** [tiangolo/fastapi#1984]
This demo generates a doc document

insert_router = APIRouter()
@insert_router.post("/items/")
async def create_item(item: Item):
    item_dict = item.dict()
    if item.tax:
        price_with_tax = item.price + item.tax
        item_dict.update({"price_with_tax": price_with_tax})
    return item_dict

This demo doesn't generate docs

@insert_router.route(path="/items/",methods=["post"])
async def create_item(item: Item):
    item_dict = item.dict()
    if item.tax:
        price_with_tax = item.price + item.tax
        item_dict.update({"price_with_tax": price_with_tax})
    return item_dict

Is this a bug?

**Comment by ycd** [comment]
Hey @xiaxichen can you reformat your code?

**Comment by SirTelemak** [comment]
`insert_router.route`  is Starlette's method that knows nothing about API. In your case you should use FastAPI method that also adds API doc `insert_router.api_route`

**Comment by xiaxichen** [comment]
> `insert_router.route` is Starlette's method that knows nothing about API. In your case you should use FastAPI method that also adds API doc `insert_router.api_route`


However, this routing registration can not be used to generate documents.
@insert_router.route(path="/items/",methods=["post"])

**Comment by xiaxichen** [comment]
`
insert_router = APIRouter()
@insert_router.post("/items/")
async def create_item(item: Item):
    item_dict = item.dict()
    if item.tax:
        price_with_tax = item.price + item.tax
        item_dict.update({"price_with_tax": price_with_tax})
    return item_dict`
`
@insert_router.route(path="/items/",methods=["post"])
    async def create_item(item: Item):
    item_dict = item.dict()
    if item.tax:
        price_with_tax = item.price + item.tax
        item_dict.update({"price_with_tax": price_with_tax})
    return item_dict`
there thanks。

**Comment by xiaxichen** [comment]
> Hey @xiaxichen can you reformat your code?
```
insert_router = APIRouter()
@insert_router.post("/items/")
async def create_item(item: Item):
	item_dict = item.dict()
	if item.tax:
		price_with_tax = item.price + item.tax
		item_dict.update({"price_with_tax": price_with_tax})
	return item_dict

This demo doesn't generate docs

@insert_router.route(path="/items/",methods=["post"])
async def create_item(item: Item):
	item_dict = item.dict()
	if item.tax:
		price_with_tax = item.price + item.tax
		item_dict.update({"price_with_tax": price_with_tax})
	return item_dict
```
there thanks

**Comment by SirTelemak** [comment]
> > `insert_router.route` is Starlette's method that knows nothing about API. In your case you should use FastAPI method that also adds API doc `insert_router.api_route`
> 
> However, this routing registration can not be used to generate documents.
> @insert_router.route(path="/items/",methods=["post"])

Yes, it wouldn't generate docs since it's starlettes method. Use
`@insert_router.api_route(path="/items/",methods=["post"])` instead

**Comment by tiangolo** [comment]
Thanks for the help here everyone! :clap: :bow: 

Thanks for reporting back and closing the issue :+1:
